### PR TITLE
Generalize E2E Tests

### PR DESCRIPTION
**Pull request by chejennifer**
- generalized E2E tests so that resources are in a per-test directory with input/ and output/.
- fixed some bugs along the way:
    - Couldn't run lint without having a tmcf file
    - The e2e tests weren't comparing the report files properly

### CODE REVIEW

File: tool/src/test/java/org/datacommons/tool/LintTest.java [hunk 1]
```diff
@@ -1,6 +1,10 @@
 package org.datacommons.tool;
 
+import java.io.File;
```
Comment (pradh) [inline tool/src/test/java/org/datacommons/tool/LintTest.java:3]: Add a short comment at the top on adding a new test-case?

**Review by pradh — APPROVED** [review]
This is neat!  So now we can just keep adding new test-cases with input/ and output/ and zero code change!
